### PR TITLE
fix: fix shortcuts when listenOn is a dialog

### PR DESCRIPTION
**Pull request by pleku**
This makes it possible for a component to specify a client side only
element that should be used for listening keydown events on when the
component is setup as the listenOn target for a Shortcut. This is needed
for components like Vaadin Dialog which transport the dialog contents
inside an overlay that only exists on the client side and thus breaks
shortcut event handling.

Related to #7799, vaadin/vaadin-dialog#229

### CODE REVIEW

File: flow-tests/test-root-context/src/main/java/com/vaadin/flow/uitest/ui/DialogShortcutView.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+    @Override
+    protected void onAttach(AttachEvent attachEvent) {
+        super.onAttach(attachEvent);
+        attachEvent.getUI().setId(UI_ID);
+    }
+
+    public void open(Dialog dialog) {
+        // final UI ui = DialogShortcutView.this.getUI().get();
+        // ui.add(this);
```
Comment (caalador) [inline flow-tests/test-root-context/src/main/java/com/vaadin/flow/uitest/ui/DialogShortcutView.java:179]: Commented out code that doesn't make sense as a comment.

File: flow-tests/test-root-context/src/main/java/com/vaadin/flow/uitest/ui/DialogShortcutView.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        add(dialog);
+    }
+
+    public void close(Dialog dialog) {
+        // final UI ui = DialogShortcutView.this.getUI().get();
+        // ui.remove(this);
```
Comment (caalador) [inline flow-tests/test-root-context/src/main/java/com/vaadin/flow/uitest/ui/DialogShortcutView.java:185]: Commented out code that doesn't make sense as a comment.

**Review by caalador — CHANGES_REQUESTED** [review]

File: flow-tests/test-root-context/src/main/java/com/vaadin/flow/uitest/ui/DialogShortcutView.java (hunk 1 above)
Comment (pleku) [reply flow-tests/test-root-context/src/main/java/com/vaadin/flow/uitest/ui/DialogShortcutView.java]: Done

File: flow-tests/test-root-context/src/main/java/com/vaadin/flow/uitest/ui/DialogShortcutView.java (hunk 2 above)
Comment (pleku) [reply flow-tests/test-root-context/src/main/java/com/vaadin/flow/uitest/ui/DialogShortcutView.java]: Done

**Review by caalador — APPROVED** [review]

**Comment by pleku** [comment]
Cherry-picked manually to 2.5 and 2.6.

**Pull request merged**